firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -242,14 +245,15 @@ public <TResult> Task<TResult> transaction(
 
   // TODO(b/261013682): Use an explicit executor in continuations.
   @SuppressLint("TaskMainThread")
-  public Task<Long> runCountQuery(Query query) {
+  public Task<Map<String, Value>> runAggregateQuery(
+      Query query, List<AggregateField> aggregateFields) {
     this.verifyNotTerminated();
-    final TaskCompletionSource<Long> result = new TaskCompletionSource<>();
+    final TaskCompletionSource<Map<String, Value>> result = new TaskCompletionSource<>();
     asyncQueue.enqueueAndForget(
         () ->
             syncEngine
-                .runCountQuery(query)
-                .addOnSuccessListener(count -> result.setResult(count))
+                .runAggregateQuery(query, aggregateFields)
```

Comment:
Another point where web/iOS are inconsistent with Android is web Converts the API AggregateField object to a model AggregateField object. This pattern is seen other places in the SDK, for example the [API Filter class](https://github.com/firebase/firebase-android-sdk/blob/aa923f7a9a578bf3c96abe4043d2b1447006b85a/firebase-firestore/src/main/java/com/google/firebase/firestore/Filter.java#L31) is converted to the [core.Filter class](https://github.com/firebase/firebase-android-sdk/blob/514402dd9f17e15f365b876143b0f139b58af258/firebase-firestore/src/main/java/com/google/firebase/firestore/core/Filter.java#L22) in the [API layer where method](https://github.com/firebase/firebase-android-sdk/blob/a9d190d4be0d69a7f85828ae572adad4c12004af/firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java#L401).

I'm not saying one way is better than the other, but I think converting to these intermediate core objects might be more consistent. This is something I'd also like @ehsannas's opinion on?

Reply:
@MarkDuckworth Are you suggesting creating an `AggregateField` class and converting the API `AggregateField` object to the model `AggregateField` object? 

Would there be any difference between these two classes? or would be a 1-to-1 mapping?

In the case of the Filter class, we needed additional functionality (methods on FieldFilter and CompositeFilter) as well as additional hierarchy (e.g. InFilter, NotInFilter, etc) that we needed internally but wasn't necessary to be seen by the external classes.

Reply:
Given your explanation, I think the implementation in Android and Java is preferred. I will backport this pattern to Web, so that SDKs are consistent.

Reply:
Actually. I'm going to leave this as is for the Web SDK. This is fine for Android. Closing.